azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageVersionServlet.java

```diff
@@ -127,59 +145,73 @@ protected void handlePost(HttpServletRequest req, HttpServletResponse resp, Sess
     }
   }
 
-  private void handleCreateImageVersion(HttpServletRequest req,
-      HttpServletResponse resp, Session session) throws ServletException, IOException {
+  private void handleCreateImageVersion(final HttpServletRequest req,
+      final HttpServletResponse resp, final Session session) throws ServletException, IOException {
     try {
-      String jsonPayload = HttpRequestUtils.getBody(req);
+      final String jsonPayload = HttpRequestUtils.getBody(req);
+      // Check for required permission to invoke the API
+      final String imageType = JSONUtils.extractTextFieldValueFromJsonString(jsonPayload, IMAGE_TYPE);
+      if (!hasImageManagementPermission(imageType, session.getUser(), Type.CREATE)) {
+        log.debug(String.format("Invalid permission to create image version "
+            + "for user: %s, image type: %s.", session.getUser().getUserId(), imageType));
+        throw new ImageMgmtInvalidPermissionException(ErrorCode.FORBIDDEN, "Invalid permission to "
```

Comment:
Why not sending the error response directly? Currently we have both, specific error classes (e.g. ImageMgmInvalidPermissionException) and error codes. Please, let's use just one approach, having both is unnecessary. In experimental APIs we're using classes, let's stick to that for consistency.

Reply:
The exact error message is formed wherever the original error is thrown and that also helps to figure out the right status code in the same place. On the other hand on the Servlet only top level exception i.e. ImageMgmtException is caught and error code and error message is sent to the response. Hence, no special handling is required.